src/webui/api/torrentscontroller.cpp

```diff
@@ -821,7 +938,31 @@ void TorrentsController::addAction()
             ? Utils::String::toEnum(contentLayoutParam, BitTorrent::TorrentContentLayout::Original)
             : std::optional<BitTorrent::TorrentContentLayout> {});
 
-    const BitTorrent::AddTorrentParams addTorrentParams
+    const DataMap &torrents = data();
+
+    QList<BitTorrent::DownloadPriority> filePriorities;
+    const QStringList filePrioritiesParam = params()[u"filePriorities"_s].split(u',', Qt::SkipEmptyParts);
+    if ((urls.size() > 1) && !filePrioritiesParam.isEmpty())
+        throw APIError(APIErrorType::BadParams, tr("You cannot specify filePriorities when adding multiple torrents"));
+    if (!torrents.isEmpty() && !filePrioritiesParam.isEmpty())
+        throw APIError(APIErrorType::BadParams, tr("You cannot specify filePriorities when uploading torrent files"));
```

Comment:
Why not? I would accept it in any case where we have single torrent metadata, either cached or just uploaded.